File: apps/java/src/main/java/chapter08/Rows.java
package chapter08;

import chapter02.Row;
import java.util.HashMap;
import java.util.Map;

/** 行を書き換えた新しい行を作る。 */
final class Rows {
  private Rows() {}

  /** 列の値を置き換えた（列が無ければ足した）新しい行を返す。元の行は変更しない。 */
  static Row with(Row row, String column, String value) {
    Map<String, String> cells = new HashMap<>(row.cells());
    cells.put(column, value);
    return new Row(cells);
  }
}

File: apps/csharp/src/MachineLearning/Chapter08/Rows.cs
namespace MachineLearning.Chapter08;

using MachineLearning.Chapter02;

/// <summary>行を書き換えた新しい行を作る。</summary>
internal static class Rows
{
    /// <summary>列の値を置き換えた（列が無ければ足した）新しい行を返す。元の行は変更しない。</summary>
    public static Row With(IReadOnlyList<string> columns, Row row, string column, string value)
    {
        var cells = columns.ToDictionary(name => name, row.Text, StringComparer.Ordinal);
        cells[column] = value;
        return new Row(cells);
    }
}
